src/controllers/activity/activity.ts

```diff
@@ -354,10 +356,12 @@ export class ActivityController extends EventEmitter {
             if (fetchTxnIdResult.status === 'rejected') {
               this.#accountsOps[selectedAccount][networkId][accountOpIndex].status =
                 AccountOpStatus.Rejected
+              updatedAccountsOps.push(this.#accountsOps[selectedAccount][networkId][accountOpIndex])
               return
             }
             if (fetchTxnIdResult.status === 'not_found') {
               declareStuckIfQuaterPassed(accountOp)
+              updatedAccountsOps.push(this.#accountsOps[selectedAccount][networkId][accountOpIndex])
```

Comment:
`declareStuckIfQuaterPassed` is called two times in `activity.ts` and we do `updatedAccountsOps.push` only at one of them, here. At code line 394 we also call `declareStuckIfQuaterPassed`.

I suggest moving `updatedAccountsOps.push` to `declareStuckIfQuaterPassed` directly and hence remove this line of code if this makes sense